### PR TITLE
Further implementation of rolling-update, add rename

### PR DESCRIPTION
**Pull request by brendandburns**
Builds on top of https://github.com/GoogleCloudPlatform/kubernetes/pull/7268

Only review the 2nd commit (a2b1983...)

@smarterclayton @bgrant0607 @jlowdermilk @ghodss

### CODE REVIEW

**Comment by smarterclayton** [comment]
@ironcladlou can you please look at the second commit in this pull?

**Comment by ironcladlou** [comment]
Makes sense to me.

**Comment by brendandburns** [comment]
@ironcladlou is that an LGTM?  Or do you want to do a further review?

**Comment by ironcladlou** [comment]
@brendanburns 
>@ironcladlou is that an LGTM? Or do you want to do a further review?

I hereby declare the looks of this PR to be: good!

**Comment by brendandburns** [comment]
rebased, @jlowdermilk, @ghodss, @bgrant0607 do you want to take a look?

**Comment by bgrant0607** [comment]
Looking now.

**Comment by davidopp** [comment]
Needs to be rebased.

**Comment by brendandburns** [comment]
Rebased.

Thanks
--brendan

**Comment by davidopp** [comment]
@brendanburns can you check the test failure?

**Comment by brendandburns** [comment]
test fixed.

Thanks!
--brendan

File: pkg/kubectl/cmd/rollingupdate.go [hunk 1]
```diff
@@ -172,10 +170,19 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 		if err != nil {
 			return err
 		}
+
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/rollingupdate.go]: The usage string (line [61](https://github.com/GoogleCloudPlatform/kubernetes/pull/7279/files#diff-57b5708f59a1279bfe3c9e55adaef5a1R61)) needs to be updated for the new syntax options.

File: pkg/kubectl/cmd/rollingupdate.go [hunk 2]
```diff
@@ -172,10 +170,19 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
...
+		var newName string
+		if len(args) >= 2 {
+			newName = args[1]
+		} else {
+			rename = true
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/rollingupdate.go]: This variable name is confusing. If rename = true, the rc will keep its original name after the update. Consider calling this e.g. "keepOldName" to better convey intended use.

File: pkg/kubectl/cmd/rollingupdate.go (hunk 1 above)
Comment (brendandburns) [inline pkg/kubectl/cmd/rollingupdate.go]: done.

File: pkg/kubectl/cmd/rollingupdate.go (hunk 2 above)
Comment (brendandburns) [inline pkg/kubectl/cmd/rollingupdate.go]: done.

**Comment by brendandburns** [comment]
@jlowdermilk ptal.

Thanks!
--brendan

File: pkg/kubectl/cmd/rollingupdate.go (hunk 1 above)
Comment (jlowdermilk) [inline pkg/kubectl/cmd/rollingupdate.go]: Updating examples good too, but I was referring specifically to this
```
		Use: "rolling-update OLD_CONTROLLER_NAME -f NEW_CONTROLLER_SPEC",
```
It should be updated to
```
		Use: "rolling-update OLD_RC_NAME ([NEW_RC_NAME] --image=IMAGE | -f NEW_RC_FILENAME)",
```

**Comment by jlowdermilk** [comment]
One remaining nit to update the `Use` string, otherwise lgtm.

File: pkg/kubectl/cmd/rollingupdate.go (hunk 1 above)
Comment (brendandburns) [inline pkg/kubectl/cmd/rollingupdate.go]: done.

**Comment by brendandburns** [comment]
Thanks for the clarification.  Updated.  ptal.

Thanks!
--brendan

**Comment by jlowdermilk** [comment]
lgtm. will merge on green.

**Comment by bgrant0607** [comment]
Filed #7402 to fix rename, assuming this is going to go in as is.

**Comment by jlowdermilk** [comment]
There's an unresolved comment in the diff for pkg/kubectl/cmd/delete.go that isn't showing up here for some reason. Please take a look and let me know if you can't see it or object to suggested change.

**Comment by bgrant0607** [comment]
Comments on individual commits (rather than on the PR) are blown away when a subsequent change (e.g., squash/rebase) is force pushed.

**Comment by jlowdermilk** [comment]
Please ignore that last comment, I hit reply in wrong window. This still lgtm.

**Pull request merged**